controller/api/public/stat_summary.go

```diff
@@ -469,6 +480,43 @@ func (s *grpcServer) getReplicationControllers(res *pb.Resource) (map[string]met
 	return rcMap, meshedPodCount, nil
 }
 
+func (s *grpcServer) getServices(res *pb.Resource) (map[string]metav1.ObjectMeta, map[string]*meshedCount, error) {
+	var err error
+	var services []*apiv1.Service
+
+	if res.Namespace == "" {
+		services, err = s.serviceLister.List(labels.Everything())
+	} else if res.Name == "" {
+		services, err = s.serviceLister.Services(res.Namespace).List(labels.Everything())
+	} else {
+		var svc *apiv1.Service
+		svc, err = s.serviceLister.Services(res.Namespace).Get(res.Name)
+		services = []*apiv1.Service{svc}
+	}
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	meshedPodCount := make(map[string]*meshedCount)
+	svcMap := make(map[string]metav1.ObjectMeta)
+	for _, svc := range services {
+		key, err := cache.MetaNamespaceKeyFunc(svc)
```

Comment:
I'm not totally clear on what this block does to calculate the meshed count.  Can you add a comment to describe this?

Reply:
yeah, this code should be DRY'd up, and comments, filed #813